tasks/release.py

```diff
@@ -1022,3 +897,396 @@ def tag_version(ctx, agent_version, commit="HEAD", verify=True, push=True, force
+    print(color_message("Preparing RC for agent version(s) {}".format(list_major_versions), "bold"))
+
+    # Step 0: checks
+
+    print(color_message("Checking repository state", "bold"))
+    ctx.run("git fetch")
+
+    if check_uncommitted_changes(ctx):
+        raise Exit(
+            color_message(
+                "There are uncomitted changes in your repository. Please commit or stash them before trying again.",
+                "red",
+            ),
+            code=1,
+        )
+
+    # Check that the current and update branches are valid
+    current_branch = ctx.run("git rev-parse --abbrev-ref HEAD", hide=True).stdout.strip()
+    update_branch = "release/{}".format(new_highest_version)
+
+    fail_if_not_base_branch(current_branch, new_highest_version)
+
+    if check_local_branch(ctx, update_branch):
+        raise Exit(
+            color_message(
+                "The branch {} already exists locally. Please remove it before trying again.".format(update_branch),
+                "red",
+            ),
+            code=1,
+        )
+
+    if check_upstream_branch(github, update_branch):
+        raise Exit(
+            color_message(
+                "The branch {} already exists upstream. Please remove it before trying again.".format(update_branch),
+                "red",
+            ),
+            code=1,
+        )
+
+    # Find milestone based on what the next final version is. If the milestone does not exist, fail.
+    milestone_name = str(next_final_version(ctx, max(list_major_versions)))
+
+    milestone = github.get_milestone_by_name(milestone_name)
+
+    if not milestone or not milestone["number"]:
+        raise Exit(
+            color_message(
+                """Could not find milestone {} in the Github repository. Response: {}
+Make sure that milestone is open before trying again.""".format(
+                    milestone_name, milestone
+                ),
+                "red",
+            ),
+            code=1,
+        )
+
+    # Step 1: Update release entries
+
+    print(color_message("Updating release entries", "bold"))
+    for major_version in list_major_versions:
+        new_version = next_rc_version(ctx, major_version, patch_version)
+        update_release_json(github.api_token, new_version)
+
+    # Step 2: Update internal module dependencies
+
+    print(color_message("Updating Go modules", "bold"))
+    update_modules(ctx, str(new_highest_version))
+
+    # Step 3: branch out, commit change, push branch
+
+    print(color_message("Branching out to {}".format(update_branch), "bold"))
+    ctx.run("git checkout -b {}".format(update_branch))
+
+    print(color_message("Committing release.json and Go modules updates", "bold"))
+    print(
+        color_message(
+            "If commit signing is enabled, you will have to make sure the commit gets properly signed.", "bold"
+        )
+    )
+    ctx.run("git add release.json")
+    ctx.run("git ls-files . | grep 'go.mod$' | xargs git add")
```

Comment:
Since we know there are no uncomitted changes I feel like it's better to just add everything (possibly checking before that only the files we expect have changed)

Reply:
I don't think we can do this, because if you `git add .` then you also add all untracked files (the files that have never been committed to the repository), which we don't want and which don't count as uncommitted changes.

Reply:
I see, makes sense if we don't want to handle the stashing (which sounds like a pain to do)